public/app/engagementindex/leaderboard/leaderboardController.js

```diff
@@ -0,0 +1,299 @@
+/**
+ * Copyright 2015 UC Berkeley (UCB) Licensed under the
+ * Educational Community License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may
+ * obtain a copy of the License at
+ *
+ *     http://opensource.org/licenses/ECL-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an "AS IS"
+ * BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+(function(angular) {
+
+  'use strict';
+
+  angular.module('collabosphere').controller('LeaderboardController', function(leaderboardFactory, userFactory, utilService, $scope) {
+
+    // Default sort
+    $scope.sortBy = 'rank';
+    $scope.reverse = true;
+
+    // Variable that will keep track of the users in this course and their points
+    $scope.users = null;
+
+    /**
+     * Get the users in the engagement index and their engagement
+     * index points
+     */
+    var getUsers = function() {
+      // The users in the engagement index can only be loaded by course admins
+      // and users that share their points with the course
+      if ($scope.me.is_admin || $scope.me.share_points) {
+        leaderboardFactory.getUsers().then(function(users) {
+          $scope.users = users;
+
+          // Extract the current user's rank
+          for (var i = 0; i < $scope.users.length; i++) {
+            if ($scope.users[i].id === $scope.me.id) {
+              $scope.me.rank = $scope.users[i].rank;
+            }
+          }
+
+          // Draw the boxplot showing how the current user ranks
+          drawBoxPlot();
+        });
+      }
+    };
+
+    /**
+     * Render the box plot showing how the current user ranks
+     * in the course
+     */
+    var drawBoxPlot = function() {
+      var points = [];
+      for (var i = 0; i < $scope.users.length; i++) {
+        points.push($scope.users[i].points);
+      }
+
+      // Wait until Angular has finished rendering the box plot container on the screen
+      setTimeout(function() {
+        // Render the box plot using highcharts
+        // @see http://api.highcharts.com/highcharts
+        chart = new Highcharts.Chart({
+          'chart': {
+            'backgroundColor': 'transparent',
+            'inverted': true,
+            // Ensure that the box plot is displayed horizontally
+            'margin': [0, 20, 0, 20],
+            'renderTo': 'leaderboard-userinfo-boxplot',
```

Comment:
@nicolaasmatthijs a suggestion would be to move this to a directive, that way you can also avoid the `setTimeout`.